bundles/mapping/mapmodule/plugin/getinfo/GetInfoPlugin.js

```diff
@@ -385,6 +387,116 @@ Oskari.clazz.define(
             me._showGfiInfo = callback;
         },
 
+        /**
+         * Sends GFIResultEvent.
+         *
+         * @method _sendGFIResultEvent
+         * @private
+         * @param  {Object} data
+         */
+        _sendGFIResultEvent: function (data) {
+            /**
+             * Checks at if data has JSON
+             * @param {Object} data
+             * @returns {Boolean} true if data has JSON Object, other false
+             */
+            const hasJSON = (data) => {
+                try {
+                    if (typeof data === 'string') {
+                        const parsedJSON = JSON.parse(data);
+
+                        // Handle non-exception-throwing cases:
+                        if ((parsedJSON && typeof parsed !== 'object') || !parsedJSON) {
+                            return false;
+                        }
+                        return true;
+                    } else if (data && typeof data === 'object') {
+                        return true;
+                    }
+                    return false;
+                } catch (e) {}
+                return false;
+            };
+            /**
+             * Checks at if data has GeoJSON
+             * @param {Object} data
+             * @returns {Boolean} true if data has GeoJSON Object, other false
+             */
+            const hasGeoJSON = (data) => {
+                try {
+                    new GeoJSON().readFeatures(data);
+                } catch (e) {
+                    return false;
+                }
+                return true;
+            };
+
+            /**
+             * Gets GFI response type
+             * @param {Object} feature
+             * @returns {String} GFI response type (json, geojson or text)
+             */
+            const getGfiResponseType = (feature) => {
+                if (hasJSON(feature.content) && !hasGeoJSON(feature.content)) {
+                    return 'json';
+                } else if (hasJSON(feature.content) && hasGeoJSON(feature.content)) {
+                    return 'geojson';
+                } else if (typeof feature === 'object' && !hasGeoJSON(feature)) {
+                    return 'json';
+                } else if (typeof feature === 'object' && hasGeoJSON(feature)) {
+                    return 'geojson';
+                }
+                return 'text';
+            };
+
+            /**
+             * Gets GFI content
+             * @param {Object} feature
+             * @returns GFI content
+             */
+            const getGfiContent = (feature) => {
+                if (hasJSON(feature.content) && typeof feature.content !== 'object') {
+                    return JSON.parse(feature.content);
```

Comment:
Here hasJSON() calls JSON.parse() only to return boolean to call JSON.parse() again on the same content.